docs/design/020-distributed-execution-and-test-suites.md

```diff
@@ -167,6 +212,81 @@ Because of the potential complexity of k6 scripts (especially ones with multiple
 
 In summary, if we make the `k6 coordinator` node dumb enough and the metric transmission efficient enough, it'd be much more likely for a test to fail because of a bad script that eats up all of the RAM, or because of network issues, or due to all sorts of other potential problems on the `agent` side... So, a non-highly-available `k6 coordinator` is probably good enough, at least to start with.
 
+#### Disruption resilience
+
+Based on the high availability assumption described, the architecture is intentionally not very highly resilient to disruption. Below, the process described for the components in the event of disruption.
+
+##### Coordinator
+
+For its nature of being a single point of failure, the `Coordinator` results to be the most impactful part of the architecture in case of disruption.
+
+In the event, Testcoordinator disappear the agent instances will not be able to complete the test as they will remain in a stuck status waiting for the connection to return reliable, after the gRPC connection's defined timeout expires, each agent instance will abort the ongoing test run and will shut down.
+
+In the event, a graceful shutdown (SIGTERM signal received) is initiated by Testcoordinator then it is expected to execute the flow described in test abortion section.
+
+##### Agent
+
+Eventually, Agent instances could be more resilient.
+
+If an instance at some point disappear, the system should tolerate this failure in some percentage requested by the user or by the default value. It means that a rendezvous point on the Sync API will have a timeout and if all the instances haven't reached the same point in a specified time frame then the Testcoordinator will remove those instances from the number of required instances for the sync consensus. 
+
+However, in a first experimental version, is not expected to implement and support this described scenario. The flow will let the instances stuck on the rendezvous point that at some point will make the request timing out and the Agent instance will start the abortion flow.
+
```

Comment:
```suggestion
For its nature of being a single point of failure, the coordinator node results to be the most impactful part of the architecture in case of disruption.

If the coordinator disappears the agent instances will not be able to complete the test as they will remain in a stuck status waiting for the connection to return reliable, after the gRPC connection's defined timeout expires, each agent instance will abort the ongoing test run and will shut down.

In the event, that a graceful shutdown (SIGTERM signal received) is initiated by the coordinator node then it is expected to execute the flow described in the test abortion section.

##### Agent

Eventually, Agent instances could be more resilient.

If an instance at some point disappears, the system should tolerate this failure in some percentage requested by the user or by the default value. It means that a rendezvous point on the Sync API will have a timeout and if all the instances haven't reached the same point in a specified time frame then the coordinator will remove those instances from the number of required instances for the sync consensus. 

However, the first experimental version is not expected to implement and support this described scenario. The flow will let the instances stuck on the rendezvous point that at some point will make the request time out and the Agent instance will start the abortion flow.
```